src/parser/core/modules/AlwaysBeCastingTable.tsx

```diff
@@ -0,0 +1,304 @@
+import {t} from '@lingui/macro'
+import {Trans} from '@lingui/react'
+import {ActionLink} from 'components/ui/DbLink'
+import Rotation from 'components/ui/Rotation'
+import {Action} from 'data/ACTIONS'
+import {Event, Events} from 'event'
+import {Analyser} from 'parser/core/Analyser'
+import {filter} from 'parser/core/filter'
+import {dependency} from 'parser/core/Injectable'
+import Downtime from 'parser/core/modules/Downtime'
+import {GlobalCooldown} from 'parser/core/modules/GlobalCooldown'
+import {Timeline} from 'parser/core/modules/Timeline'
+import React from 'react'
+import {Table, Button, Message} from 'semantic-ui-react'
+import {EventHook} from '../Dispatcher'
+import {Data} from './Data'
+import DISPLAY_ORDER from './DISPLAY_ORDER'
+
+//value to be added to the gcd to avoid false positives. 100ms for caster tax, 50ms for gcd jitter.
+const GCD_JITTER_OFFSET = 50
+const GCD_CASTER_TAX_OFFSET = 100
+
+const BASE_RECAST = 2500
+const OGCD_OFFSET = 1000
+
+interface Window {
+	start: number,
+	startAction: Events['action'],
+	stop?: number,
+	stopAction?: Events['action'],
+	recastTime: number,
+	doNothingForegivness: number, //allowances based on time dead, interruptions, oGCDs
+	casterTax: number,
+	deathCount?: number,
+	interruptedActions?: Action[],
+	actions: Array<Events['action']>, //used to track oGCDs for weaves
+}
+
+export class ABCTable extends Analyser {
+	static override handle = 'alwaysbecastingtable'
```

Comment:
Since this is output related to AlwaysBeCasting, it probably should just be in AlwaysBeCasting, rather than a separate Analyser

Reply:
sounds good! I'll look at combining the two into one Analyser (maybe after this one is done so I don't create any funny business in abc)